test/e2e/invalid_test.go

```diff
@@ -78,6 +78,72 @@ func TestInvalidBuild(t *testing.T) {
 			Steps:   []corev1.Container{{Image: "busybox"}},
 			Timeout: &metav1.Duration{time.Duration(36 * time.Hour)},
 		},
+	}, {
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: buildTestNamespace,
+			Name:      "source-with-duplicate-subpath",
```

Comment:
I don't understand the problem with this testcase.

I would expect specifying two sources, one going to `paht/foo` and the other `path/foo` should work since `paht` != `path`